api/src/pcapi/core/offers/models.py

```diff
@@ -354,6 +354,8 @@ class Offer(PcObject, Model, ExtraDataMixin, DeactivableMixin, ProvidableMixin):
     #  can be used by PostgreSQL when filtering on the `venueId` column only.
     Index("venueId_idAtProvider_index", venueId, idAtProvider, unique=True)
 
+    Index("offer_expr_idx", ExtraDataMixin.extraData["isbn"].astext)
```

Comment:
Même si cet index existe sur quelques environnements, une migration `CREATE INDEX CONCURRENTLY IF NOT EXISTS` correspondante me semble nécessaire.

Aussi, on peut en profiter  pour ajouter dans un commit séparé un index des `isbn` de la table `product`.